### ISSUE TITLE
Progress for Path Follow Node does not update position/globalPosition

### ISSUE THREAD

**Issue by EricXu1728** [godotengine/godot#100719]
### Tested versions

- Reproducible in 4.3 official release and 4.4-dev7
-  Not reproducible in 4.3 dev 2 and behind. 
-  Did not test versions between 4.3 and 4.3 dev 2

### System information

Windows 11 ulkan API 1.3.280 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 4090

### Issue description

```gdscript 
extends PathFollow3D

func _ready():
	var i = 10;
	
	for n in i:
		progress = (100/i) * n;
		
		self.progress_ratio = progress
		print(self.progress)
		print(self.transform)
		
	
	pass 
```

Expected behavior: As you update the progress of a path, it will update it's transform:
```
Godot Engine v4.3.dev2.mono.official.352434668 - https://godotengine.org
Vulkan API 1.3.280 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 4090
 
0
[X: (0.460565, -0, 0.887626), Y: (0.03469, 0.999236, -0.018), Z: (-0.886948, 0.039081, 0.460213), O: (-2.73196, 0.481295, 1.19287)]
13.9340238571167
[X: (0.247949, 0.059038, 0.966973), Y: (0.42345, 0.891137, -0.162987), Z: (-0.871327, 0.449877, 0.195957), O: (9.512231, -3.07611, -3.28165)]
0.91930961608887
[X: (0.460565, 0, 0.887626), Y: (0.03469, 0.999236, -0.018), Z: (-0.886948, 0.039081, 0.460213), O: (-1.916613, 0.445368, 0.769808)]
```
etc...

Current behavior:
```
Godot Engine v4.3.stable.mono.official.77dcf97d8 - https://godotengine.org
Vulkan 1.3.280 - Forward+ - Using Device #0: NVIDIA - NVIDIA GeForce RTX 4090

0
[X: (0.460565, -0, 0.887626), Y: (0.03469, 0.999236, -0.018), Z: (-0.886948, 0.039081, 0.460213), O: (-2.73196, 0.481295, 1.19287)]
13.9340238571167
[X: (0.460565, -0, 0.887626), Y: (0.03469, 0.999236, -0.018), Z: (-0.886948, 0.039081, 0.460213), O: (-2.73196, 0.481295, 1.19287)]
0.91930961608887
[X: (0.460565, -0, 0.887626), Y: (0.03469, 0.999236, -0.018), Z: (-0.886948, 0.039081, 0.460213), O: (-2.73196, 0.481295, 1.19287)]
1.83862686157227
```

### Steps to reproduce

Check a path follow node's position
Change the position
Check the position again

### Minimal reproduction project (MRP)

[bugs.zip](https://github.com/user-attachments/files/18220103/bugs.zip)

**Comment by AThousandShips** [comment]
I assume that the progress updates when processing either in `_process` or in `_physics_process`, try doing this outside of `_ready`

**Comment by EricXu1728** [comment]
This also does not work in the process:

```godot
extends PathFollow3D

var has_processed = false

func _process(delta):
	if not has_processed:
		has_processed = true 
		
		var i = 10
		for n in i:
			var progress = (100 / i) * n
			self.progress_ratio = progress
			print(self.progress)
			print(self.transform)

```
In fact, it introduces new weirder error where self.progress is always the minimum or maximum length of the curve
```
Godot Engine v4.3.stable.mono.official.77dcf97d8 - https://godotengine.org
Vulkan 1.3.280 - Forward+ - Using Device #0: NVIDIA - NVIDIA GeForce RTX 4090

0
[X: (0.460565, -0, 0.887626), Y: (0.03469, 0.999236, -0.018), Z: (-0.886948, 0.039081, 0.460213), O: (-2.73196, 0.481295, 1.19287)]
13.9340238571167
[X: (0.460565, -0, 0.887626), Y: (0.03469, 0.999236, -0.018), Z: (-0.886948, 0.039081, 0.460213), O: (-2.73196, 0.481295, 1.19287)]
13.9340238571167

```

It does behave normally if it isn't done all at once, but this is still restrictive of what can be done with the follow node:
```godot
extends PathFollow3D

var i = 10
var n = 0  

func _process(delta):
	if n < i:
		var progress = (100 / i) * n
		self.progress_ratio = progress
		print("Progress:", self.progress_ratio)
		print("Transform:", self.transform)
		n += 1 
```

**Comment by EricXu1728** [comment]
> I assume that the progress updates when processing either in `_process` or in `_physics_process`, try doing this outside of `_ready`

This also does not work in the process:

```godot
extends PathFollow3D

var has_processed = false

func _process(delta):
	if not has_processed:
		has_processed = true 
		
		var i = 10
		for n in i:
			var progress = (100 / i) * n
			self.progress_ratio = progress
			print(self.progress)
			print(self.transform)

```
In fact, it introduces new weirder error where self.progress is always the minimum or maximum length of the curve
```
Godot Engine v4.3.stable.mono.official.77dcf97d8 - https://godotengine.org
Vulkan 1.3.280 - Forward+ - Using Device #0: NVIDIA - NVIDIA GeForce RTX 4090

0
[X: (0.460565, -0, 0.887626), Y: (0.03469, 0.999236, -0.018), Z: (-0.886948, 0.039081, 0.460213), O: (-2.73196, 0.481295, 1.19287)]
13.9340238571167
[X: (0.460565, -0, 0.887626), Y: (0.03469, 0.999236, -0.018), Z: (-0.886948, 0.039081, 0.460213), O: (-2.73196, 0.481295, 1.19287)]
13.9340238571167

```

It does behave normally if it isn't done all at once, but this is still restrictive of what can be done with the follow node:
```godot
extends PathFollow3D

var i = 10
var n = 0  

func _process(delta):
	if n < i:
		var progress = (100 / i) * n
		self.progress_ratio = progress
		print("Progress:", self.progress_ratio)
		print("Transform:", self.transform)
		n += 1 
```

**Comment by AThousandShips** [comment]
I'm talking about updating the progress multiple times at once, try doing *one* update per frame

**Comment by EricXu1728** [comment]
Would this be considered expected behavior moving forward and not a bug then? I feel like this would interfere with some reasonable uses of the path node.

**Comment by AThousandShips** [comment]
That's what's to be discussed, the PR changing this was approved and merged with the awareness of this change so it's probably expected behavior, and so something to document

**Comment by AThousandShips** [comment]
Seems part of the change was considered a regression, see:
* https://github.com/godotengine/godot/issues/95612

Solved in:
* https://github.com/godotengine/godot/pull/96140

Which has been cheery picked to 4.3.1 which will be released in the future 

Closing this as the issue is solved in 4.4 and will be in the next release
